experimental/streaming/exporter/reader/reader.go

```diff
@@ -265,22 +222,32 @@ func (ro *readerObserver) orderedObserve(event observer.Event) {
 			read.SpanContext = span.spanContext
 		}
 
-	case observer.RECORD_STATS:
-		read.Type = RECORD_STATS
+	case observer.UPDATE_METRIC:
+		read.Type = event.Type
 
-		_, span := ro.readScope(event.Scope)
-		if span != nil {
-			read.SpanContext = span.spanContext
+		if event.Context != nil {
+			span := trace.CurrentSpan(event.Context)
+			if span != nil {
+				read.SpanContext = span.SpanContext()
+			}
 		}
-		for _, es := range event.Stats {
-			ro.addMeasurement(&read, es)
+		ro.addMeasurement(&read, event.Measurement)
+
+	case observer.BATCH_METRIC:
+		read.Type = event.Type
+
+		if event.Context != nil {
+			span := trace.CurrentSpan(event.Context)
+			if span != nil {
```

Comment:
Same here, I think.

Reply:
Yeah, just a nil check looks good. If the reader will panic because SpanContext isn't set, let's fix that.